go/DEVELOPER.md

```diff
@@ -0,0 +1,206 @@
+
+### Build from source
+
+#### Prerequisites
+
+Software Dependencies
+
+-   Go
+-   GNU Make
+-   git
+-   GCC
+-   pkg-config
+-   protoc (protobuf compiler) >= v3.20.0
+-   openssl
+-   openssl-dev
+-   rustup
+
+
+**Dependencies installation for Ubuntu**
+
+```bash
+sudo apt update -y
+sudo apt install -y git gcc pkg-config openssl libssl-dev unzip make
+# Install Go
+sudo snap install go --classic
+export PATH="$PATH:$HOME/go/bin"
+# Install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+# Install protobuf compiler
+PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+curl -LO $PB_REL/download/v3.20.3/protoc-3.20.3-linux-x86_64.zip
+unzip protoc-3.20.3-linux-x86_64.zip -d $HOME/.local
+export PATH="$PATH:$HOME/.local/bin"
+# Check that the protobuf compiler is installed
+protoc --version
+```
+
+**Dependencies installation for CentOS**
+
+```bash
+sudo yum update -y
+sudo yum install -y git gcc pkgconfig openssl openssl-devel unzip wget tar
+# Install Go
+wget https://go.dev/dl/go1.22.0.linux-amd64.tar.gz
+sudo tar -C /usr/local -xzf go1.22.0.linux-amd64.tar.gz
+export PATH="$PATH:/usr/local/go/bin"
+export PATH="$PATH:$HOME/go/bin"
+# Install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+# Install protobuf compiler
+PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+curl -LO $PB_REL/download/v3.20.3/protoc-3.20.3-linux-x86_64.zip
+unzip protoc-3.20.3-linux-x86_64.zip -d $HOME/.local
+export PATH="$PATH:$HOME/.local/bin"
+# Check that the protobuf compiler is installed
+protoc --version
+```
+
+**Dependencies installation for MacOS**
+
+```bash
+brew update
+brew install go make git gcc pkgconfig protobuf@3 openssl
+export PATH="$PATH:$HOME/go/bin"
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+```
+
+#### Building and installation steps
+
+Before starting this step, make sure you've installed all software requirements.
+
+1. Clone the repository:
+    ```bash
+    VERSION=0.1.0 # You can modify this to other released version or set it to "main" to get the unstable branch
+    git clone --branch ${VERSION} https://github.com/aws/glide-for-redis.git
+    cd glide-for-redis
+    ```
+2. Initialize git submodule:
+    ```bash
+    git submodule update --init --recursive
+    ```
+3. Install build dependencies:
+    ```bash
+    cd go
+    make install-build-tools
+    ```
+4. Build the Go wrapper:
+    ```bash
+    make build
+    ```
+5. Run tests:
+    1. Ensure that you have installed redis-server and redis-cli on your host. You can find the Redis installation guide at the following link: [Redis Installation Guide](https://redis.io/docs/install/install-redis/install-redis-on-linux/).
```

Comment:
add redis for prerequisite list above

Reply:
It looks like the node/python DEVELOPER.md files do not include it in the prerequisites, should I add it anyways?

Reply:
Yes